tests/attention/test_sparse_mla_sm120.py

```
@@ -0,0 +1,908 @@
+    indices: torch.Tensor,
+    sm_scale: float,
+    d_v: int,
+    attn_sink: torch.Tensor | None = None,
+    topk_length: torch.Tensor | None = None,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    """Dense SDPA over the sparse-gathered KV. Returns (output_bf16, lse_log2).
+
+    Honors ``attn_sink`` (FlashMLA V4 sink-merge convention) and
+    ``topk_length`` (per-token valid-length mask).
+    """
+    num_tokens, num_heads, d_qk = q.shape
```

Ruff (0.15.13)

[warning] Unpacked variable `num_heads` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)